### PR TITLE
Update lr_scheduler.py to check the type of eta_min

### LINKED ISSUE #87757: CosineAnnealingWarmRestarts is not throwing error when arguments are of different types
### 🐛 Describe the bug

One of the parameters for the scheduler [CosineAnnealingWarmRestarts](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CosineAnnealingWarmRestarts.html) is `eta_min`. While `eta_min` is an optional argument, the default is `float` 0. However, if you pass `eta_min` as any string, the scheduler doesn't throw an error, nor gives the warning. 

I discovered this, when I was using a config file to pass all the arguments and for some strange reason arguments were read as strings. Hence,  `eta_min = '1e-6'`. 

Here's small code snippet to reproduce:
```
import matplotlib.pyplot as plt
import torch

model = torch.nn.Linear(2,1)
optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
scheduler = torch.optim.lr_scheduler.CosineAnnealingWarmRestarts(optimizer, T_0=10, T_mult=2, eta_min='1e-6')

lrs = []
lrs_opt = []
for _ in range(100):
    optimizer.zero_grad()
    optimizer.step()
    scheduler.step()
    lrs_sched.append(scheduler.get_last_lr())
    lrs_opt.append(optimizer.param_groups[0]["lr"])

plt.plot(lrs_sched, color='green',)
plt.show()
plt.plot(lrs_opt, color='red')
plt.show()
 ```
Replace `eta_min='1e-6'` to `eta_min='any_rubbish_string'`, the code still works and gives a constant learning rate of 0.01.

This is my first time opening an issue in this repo, please let me know if more information is required.
 

### Versions

Collecting environment information...
PyTorch version: 1.13.0.dev20220707
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (conda-forge gcc 12.1.0-16) 12.1.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.15.0-50-generic-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: 10.1.243
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3090
Nvidia driver version: 470.141.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] torch==1.13.0.dev20220707
[pip3] torchvision==0.14.0.dev20220707
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               h2bc3f7f_2  
[conda] libblas                   3.9.0            12_linux64_mkl    conda-forge
[conda] libcblas                  3.9.0            12_linux64_mkl    conda-forge
[conda] liblapack                 3.9.0            12_linux64_mkl    conda-forge
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py38h7f8727e_0  
[conda] mkl_fft                   1.3.1            py38hd3c417c_0  
[conda] mkl_random                1.2.2            py38h51133e4_0  
[conda] numpy                     1.22.3           py38he7a7128_0  
[conda] numpy-base                1.22.3           py38hf524024_0  
[conda] pytorch                   1.13.0.dev20220707 py3.8_cuda11.3_cudnn8.3.2_0    pytorch-nightly
[conda] pytorch-mutex             1.0                        cuda    pytorch-nightly
[conda] torchvision               0.14.0.dev20220707      py38_cu113    pytorch-nightly

### PR DESCRIPTION
**Pull request by ravikiranrao**
Add float assertion to `eta_min` parameter in `CosineAnnealingWarmRestarts`.

Fixes #87757

### CODE REVIEW

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -1367,6 +1367,8 @@ def __init__(self, optimizer, T_0, T_mult=1, eta_min=0, last_epoch=-1, verbose=F
             raise ValueError("Expected positive integer T_0, but got {}".format(T_0))
         if T_mult < 1 or not isinstance(T_mult, int):
             raise ValueError("Expected integer T_mult >= 1, but got {}".format(T_mult))
+        if not isinstance(eta_min, float):
+            raise ValueError("Exprected float, but got {}".format(eta_min))
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1371]:
```suggestion
            raise ValueError("Expected float eta_min, but got {} of type {}".format(eta_min, type(eta_min))
```
nit with spelling, and give a better error message if the type is string since it’d be hard to tell without printing the type if someone passes in “1e-5”.

**Review by janeyx99 — COMMENTED** [review]
Thank you for the change! I left a comment—once addressed, this should be good to go!

File: torch/optim/lr_scheduler.py (hunk 1 above)
Comment (ravikiranrao) [reply torch/optim/lr_scheduler.py:1371]: Ah! Sorry for the typo. Thanks for having `type` in the error message. This makes more sense. @janeyx99

**Comment by ravikiranrao** [comment]
> Thank you for the change! I left a comment—once addressed, this should be good to go!

Addressed @janeyx99

**Comment by janeyx99** [comment]
alright, approving CI. let’s ensure it’s all green—in the meantime, could you update the PR title to be more descriptive, such as mentioning that you’re adding error checking for eta_min type in lr scheduler?

**Comment by janeyx99** [comment]
CI failures are legit, looks like we’re missing a )

**Comment by ravikiranrao** [comment]
> CI failures are legit, looks like we’re missing a )

Done.

**Comment by ravikiranrao** [comment]
CI failures show that the assertion fails when `eta_min` is `int` type. Fixing this.

**Comment by ravikiranrao** [comment]
cc @albanD.

**Review by janeyx99 — APPROVED** [review]
lgtm!
